docs/rfc/rfc-1.adoc

```diff
@@ -0,0 +1,143 @@
+address to the main wallet address, which allows the system to mint tBTC and
+disables the 30-day refund.
+
+==== Automated Refunds
+
+A bitcoin transaction is an amount and a script. The script can be something as
+simple as "these funds can be spent by wallet 0xabc", or in our case, as
+complex as "these funds can be spent by wallet 0xabc but if they aren't spent
+within 30 days they can be spent by wallet 0x123". This gives us the ability to
+create deposits that automatically are refunded after 30 days if they aren't
+swept. Thus, if a user misfunds or they get cold feet (for any reason), all
+they need to do is not submit their reveal and wait 30 days.
+
+=== Redemption
+
+To initiate a redemption, a user supplies an amount `x` of TBTC and a bitcoin
+address. Then, the system calculates the redemption fee `fee`, and releases an
+amount of bitcoin `y` such that `x = y + fee` to the supplied bitcoin address.
+`y` amount of TBTC is burned to maintain the peg. The remaining `fee` TBTC is
+sold by the system to buy back `T` tokens (more about this process in the fee
+section).
+
+In the MVP version of the system, a redemption is capped at the amount of
+bitcoin contained in the largest wallet. Redemptions are fulfilled from the
+oldest wallet that contains enough bitcoin to fulfil the redemption. If more
+tBTC needs to be redeemed than there is in the largest wallet, then the user
+needs to submit multiple redemptions. After a redemption, if a wallet has under
+a governable threshhold of BTC remaining, it transfers that BTC to the active
+wallet and closes.
+
+=== Wallet Lifecycle
+
+Wallets are periodically created, where the period length is a governable
+parameter. To create a new wallet, a group of 100 operators is selected from
+the pool of available operators (some operators may be selected twice if there
+are not enough) using a process called sortition. The probabiliy that a
+particular operator is chosen is based on their stake weight, which in turn is
+based on the number of `T` tokens they have invested in the staking contract.
+
+Once the 100 operators have been selected, they generate a 51-of-100 ecdsa
+signing group to handle the bitcoin key material. The distributed key
+generation process requires that all 100 participants are available, but future
+signing events (like minting and redemption) only require 51 of the 100.
+
+As time passes and operators drop out of the system, a wallet becomes at risk of
+being able to meet the 51-of-100 threshhold to produce signatures. Additionally,
+we want to avoid situations where operators are the custodians of a wallet for
+extended periods. To avoid these issues, we can set a max age of a wallet and a minimum
+liveness threshhold. Once a wallet is older than the max age, or if it drops below
+the liveness threshhold (say, below 60 on a heartbeat), we motion to transfer
+the funds to another randomly selected wallet.
```

Comment:
How are we selecting that random wallet? Wouldn't it be simpler to always transfer to the most recent wallet?

Reply:
> How are we selecting that random wallet?

That's a good question - calling the random beacon might be too expensive. I think the recent-wallet strategy is the easiest and least costly (because the newest wallet should be the one with the highest expected lifespan), but it makes us a little more vulnerable to folks intentionally trying to gain control over the newest wallet and then trying to create a transfer from an older wallet to the wallet they now control.

Reply:
Can we choose the next random wallet by seeding a RNG based on the hash of the next ethereum block? We don't need a *ton* of security here imo